src/app/applications/applications.controller.app.js

```diff
@@ -47,9 +66,45 @@
     }
 
     function uploadIcon(appName) {
-      rClient.uploadIcon(appName).then(function() {
-          $window.location = '#/app/' + appName + '/overview';
+      $scope.cropImage(function() {
+          var dataUrl = $scope.cropImage.dataUrl;
+          var croppedBlob = Cropper.decode(dataUrl);
+          var croppedImage = new File([croppedBlob], $scope.filename);
+          var formObj = new FormData();
+
+          formObj.append('file', croppedImage);
+          rClient.uploadIcon(appName, formObj).then(function() {
+            $window.location.reload(true);
+          });
+      });
+    }
+
+    function selectFile() {
+      $('input[id=file-id]').click();
+    }
+
+    function onFile() {
+      var input = $('#file-id')[0];
+      $scope.fileToCrop = input.files[0];
+      Cropper.encode($scope.fileToCrop).then(function(dataUrl) {
+        $scope.dataUrl = dataUrl;
+        $timeout(showCropper);
+        $scope.showButton = true;
       });
     }
+
+    function cropImage(callback) {
+      if (!$scope.fileToCrop || !$scope.data) { return;}
```

Comment:
Please, avoid these inline blocks in JS code.
Gulp has a module to shorten the code. So, don't worry about reducing the code's size.